File: nagasenacs/src/Nagasena/Proc/IO/Compression/ScriberChannelFactory.cs
using Channel = Nagasena.Proc.Common.Channel;

namespace Nagasena.Proc.IO.Compression {

  internal sealed class ScriberChannelFactory : ChannelFactory {

    internal override Channel createChannel(int firstPos, int blockNum) {
      return new ScriberChannel(firstPos, blockNum);
    }

  }

}

File: sumerogi/src/com/sumerogi/proc/io/compression/ScriberChannelFactory.java
package com.sumerogi.proc.io.compression;

import com.sumerogi.proc.common.Channel;

final class ScriberChannelFactory extends ChannelFactory {

  @Override
  public Channel createChannel(int firstPos, int blockNum) {
    return new ScriberChannel(firstPos, blockNum);
  }

}
